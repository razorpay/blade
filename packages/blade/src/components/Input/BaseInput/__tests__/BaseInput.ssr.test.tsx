import { BaseInput } from '..';
import renderWithSSR from '~utils/testing/renderWithSSR.web';

describe('<BaseInput />', () => {
  it('should display success validation state', () => {
    const label = 'Enter name';
    const { getByText, getByLabelText, container } = renderWithSSR(
      <BaseInput
        label={label}
        id="name"
        validationState="success"
        successText="Success"
        helpText="Help"
        errorText="Error"
      />,
    );

    const input = getByLabelText(label);
    const successText = getByText('Success');

    expect(successText).toBeTruthy();
    expect(input).toHaveAccessibleDescription('Success');
    expect(input).toBeValid();
    expect(container).toMatchSnapshot();
  });
});
