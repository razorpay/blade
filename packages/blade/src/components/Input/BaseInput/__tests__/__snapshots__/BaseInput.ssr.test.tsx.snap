// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> should display success validation state 1`] = `"<div id="root"><div data-blade-component="base-box" class="BaseBox-bmPWx"><div data-blade-component="base-box" class="BaseBox-bmPWx hqusWt"><div data-blade-component="base-box" class="BaseBox-bmPWx jaAmIu"><label for="name-undefined-input-undefined" style="width:auto;flex-shrink:0;margin-right:0px" data-blade-component="form-label"><div data-blade-component="base-box" class="BaseBox-bmPWx fZlcZT"><div data-blade-component="base-box" class="BaseBox-bmPWx ikPXAQ"><p class="StyledBaseText-dVBfTO bDCeGT" data-blade-component="text">Enter name</p><div data-blade-component="visually-hidden" class="VisuallyHiddenweb__StyledVisuallyHidden-g3hls3-0 Fuolv"><p class="StyledBaseText-dVBfTO fAQkza" data-blade-component="text"></p></div></div></div></label></div><div data-blade-component="base-box" class="BaseBox-bmPWx BaseInput__FocusRingWrapper-sc-177qd3e-0  iIoIGQ"><div data-blade-component="base-box" class="BaseBox-bmPWx AnimatedBaseInputWrapperweb__StyledBaseInputWrapper-e1vobd-0 AnimatedBaseInputWrapperweb__StyledAnimatedBaseInputWrapper-e1vobd-1  kjPsHx"><input type="text" id="name-undefined-input-undefined" data-blade-component="styled-base-input" aria-required="false" aria-disabled="false" aria-invalid="false" aria-describedby="name-undefined-successtext-undefined" class="StyledBaseInputweb__StyledBaseNativeInput-hsusrk-0 ickwiI"/></div></div></div><div data-blade-component="base-box" class="BaseBox-bmPWx iARoiz"><div data-blade-component="base-box" class="BaseBox-bmPWx ePSvGO"><div id="name-undefined-successtext-undefined" data-blade-component="base-box" class="BaseBox-bmPWx kzOBQv"><span style="word-break:break-all" data-blade-component="base-box" class="BaseBox-bmPWx fAEwZa"><svg aria-hidden="true" data-blade-component="icon" height="12px" viewBox="0 0 24 24" width="12px" fill="none" class="Svgweb__StyledSvg-vcmjs8-0"><path d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.31658 18.0976 8.68342 18.0976 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z" clip-rule="evenodd" fill="hsla(150, 100%, 27%, 1)" fill-rule="evenodd" data-blade-component="svg-path"></path></svg><div data-blade-component="base-box" class="BaseBox-bmPWx eLBOzs"></div><span class="StyledBaseText-dVBfTO jnVJao" data-blade-component="text">Success</span></span></div></div></div></div></div>"`;

exports[`<BaseInput /> should display success validation state 2`] = `
.c0.c0.c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
}

.c1.c1.c1.c1.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0px;
  margin-top: 0px;
}

.c2.c2.c2.c2.c2 {
  margin-bottom: 4px;
}

.c3.c3.c3.c3.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-height: 36px;
  gap: 0px;
}

.c10.c10.c10.c10.c10 {
  margin-left: 0px;
}

.c11.c11.c11.c11.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c12.c12.c12.c12.c12 {
  margin-top: 4px;
}

.c13.c13.c13.c13.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14.c14.c14.c14.c14 {
  margin-right: 4px;
}

.c4.c4.c4.c4.c4 {
  color: hsla(211,26%,34%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.75rem;
  font-weight: 600;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
  overflow: hidden;
  display: -webkit-box;
  line-clamp: 2;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.c6.c6.c6.c6.c6 {
  color: hsla(212,39%,16%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c15.c15.c15.c15.c15 {
  color: hsla(150,100%,27%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.6875rem;
  font-weight: 400;
  font-style: italic;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c9.c9.c9.c9.c9 {
  color: hsla(211,26%,34%,1);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: transparent;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 12px;
  padding-right: 12px;
  width: 100%;
  height: 36px;
  min-height: 36px;
  resize: none;
  outline: none;
  border: none;
  cursor: auto;
}

.c9.c9.c9.c9.c9::-webkit-input-placeholder {
  color: hsla(211,20%,52%,0.32);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c9.c9.c9.c9.c9::-moz-placeholder {
  color: hsla(211,20%,52%,0.32);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c9.c9.c9.c9.c9:-ms-input-placeholder {
  color: hsla(211,20%,52%,0.32);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c9.c9.c9.c9.c9::placeholder {
  color: hsla(211,20%,52%,0.32);
  font-family: "Inter","Inter Fallback Arial",Arial;
  font-size: 0.875rem;
  font-weight: 400;
  font-style: normal;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin: 0;
  padding: 0;
}

.c9.c9.c9.c9.c9:focus {
  outline: none;
}

.c8.c8.c8.c8.c8 {
  background-color: hsla(0,0%,100%,1);
  border-radius: 4px;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  position: relative;
  border: none;
  border-width: 0px;
  box-shadow: hsla(150,100%,32%,1) 0px 0px 0px 1.5px;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-easing: cubic-bezier(0.3,0,0.2,1);
  transition-easing: cubic-bezier(0.3,0,0.2,1);
}

.c8.c8.c8.c8.c8:hover {
  background-color: hsla(210,40%,98%,1);
  border-radius: 4px;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  position: relative;
  border: none;
  border-width: 0px;
  box-shadow: hsla(150,100%,32%,1) 0px 0px 0px 1.5px;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-easing: cubic-bezier(0.3,0,0.2,1);
  transition-easing: cubic-bezier(0.3,0,0.2,1);
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
}

.c8.c8.c8.c8.c8:focus-within {
  background-color: hsla(0,0%,100%,1);
  border-radius: 4px;
  border-style: solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  position: relative;
  border: none;
  border-width: 0px;
  box-shadow: hsla(150,100%,32%,1) 0px 0px 0px 1.5px;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-easing: cubic-bezier(0.3,0,0.2,1);
  transition-easing: cubic-bezier(0.3,0,0.2,1);
}

.c5.c5.c5.c5.c5 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: 0 -1px -1px 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  left: -10000px;
  white-space: nowrap;
  word-wrap: normal;
}

.c7.c7.c7.c7.c7 {
  border-radius: 4px;
  width: 100%;
}

.c7.c7.c7.c7.c7:focus-within {
  outline: 4px solid hsla(227,100%,59%,0.18);
  outline-offset: 1px;
  -webkit-transition-property: outline-width;
  transition-property: outline-width;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-timing-function: cubic-bezier(0.3,0,0.2,1);
  transition-timing-function: cubic-bezier(0.3,0,0.2,1);
}

<div
  id="root"
>
  <div
    class=""
    data-blade-component="base-box"
  >
    <div
      class="c0"
      data-blade-component="base-box"
    >
      <div
        class="c1"
        data-blade-component="base-box"
      >
        <label
          data-blade-component="form-label"
          for="name-1-input-2"
          style="width:auto;flex-shrink:0;margin-right:0px"
        >
          <div
            class="c2"
            data-blade-component="base-box"
          >
            <div
              class="c3"
              data-blade-component="base-box"
            >
              <p
                class="c4"
                data-blade-component="text"
              >
                Enter name
              </p>
              <div
                class="c5"
                data-blade-component="visually-hidden"
              >
                <p
                  class="c6"
                  data-blade-component="text"
                />
              </div>
            </div>
          </div>
        </label>
      </div>
      <div
        class=" c7"
        data-blade-component="base-box"
      >
        <div
          class="AnimatedBaseInputWrapperweb__StyledAnimatedBaseInputWrapper-e1vobd-1  c8"
          data-blade-component="base-box"
        >
          <input
            aria-describedby="name-1-successtext-5"
            aria-disabled="false"
            aria-invalid="false"
            aria-required="false"
            class="c9"
            data-blade-component="styled-base-input"
            id="name-1-input-2"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="c10"
      data-blade-component="base-box"
    >
      <div
        class="c11"
        data-blade-component="base-box"
      >
        <div
          class="c12"
          data-blade-component="base-box"
          id="name-1-successtext-5"
        >
          <span
            class="c13"
            data-blade-component="base-box"
            style="word-break:break-all"
          >
            <svg
              aria-hidden="true"
              class="Svgweb__StyledSvg-vcmjs8-0"
              data-blade-component="icon"
              fill="none"
              height="12px"
              viewBox="0 0 24 24"
              width="12px"
            >
              <path
                clip-rule="evenodd"
                d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.31658 18.0976 8.68342 18.0976 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                data-blade-component="svg-path"
                fill="hsla(150, 100%, 27%, 1)"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="c14"
              data-blade-component="base-box"
            />
            <span
              class="c15"
              data-blade-component="text"
            >
              Success
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
